### PR TITLE
Remove python cryptography libraries in favor of subprocess to `openssl`

### PR DESCRIPTION
**Pull request by addyess**
Trying to deploy this charm to arm64 resulted in something fairly broken

Much of it was the fact that charmcraft packs binary packages into the charm for cryptography.  That library was driven by the charm's need to generate a self-signed cert.  

So -- this PR addresses the inclusion of cryptography by using openssl and subprocess to generate certs in an arch agnostic way.

### CODE REVIEW

File: charmcraft.yaml [hunk 1]
```diff
@@ -3,16 +3,20 @@
...
   charm:
     charm-python-packages: [setuptools, pip]
     build-packages:
       - git
-      # required for installing cffi:
-      - libffi-dev
-      # required for installing cryptography (cargo also contains rust):
-      - cargo
-      - libssl-dev
-      - pkg-config
```
Comment (addyess) [inline charmcraft.yaml:18]: all these are now unnecessary

File: charmcraft.yaml [hunk 2]
```diff
@@ -3,16 +3,20 @@
...
+      channel: "20.04"
+    run-on:
+    - name: ubuntu
+      channel: "20.04"
+      architectures:
+        - amd64
+        - arm64
+        - armhf
+        - ppc64
+        - s390x
```
Comment (addyess) [inline charmcraft.yaml:17]: represents all the supported architectures of upstream kubernetes-dashboard

**Comment by jnsgruk** [comment]
I might be tempted to remove this functionality completely and just implement the `tls-certificates` interface - that way the charm will be able to get certs from vault, or statically from the `tls-certificates-operator`

**Comment by addyess** [comment]
@jnsgruk yes, i will do that in a following PR.  I have the code prepared for that, just wanted to wait for a separate PR

**Review by kwmonroe — APPROVED** [review]
Nice refactor to keep the self-signed option open for folks.  Might even pack a bit faster too.

**Pull request merged**